### DISCUSSION TITLE
Client 1 quit with exit code 1 and signal 0

### DISCUSSION THREAD

**Discussion by wviana** [mfussenegger/nvim-jdtls#162] [Q&A] [answered]
Hi there,

I'm trying to configure my nvim-jdtls, but I'm getting a `Client 1 quit with exit code 1 and signal 0`

Here is my java.lua

```lua
local workspace_dir = vim.fn.fnamemodify(vim.fn.getcwd(), ':p:h:t')
local config = {
  cmd = {
	'java',
        '-Declipse.application=org.eclipse.jdt.ls.core.id1',
        '-Dosgi.bundles.defaultStartLevel=4',
        '-Declipse.product=org.eclipse.jdt.ls.core.product',
        '-Dlog.level=ALL',
        '-noverify',
        '-Xmx1G',
        '-jar ~/Apps/jdtls/plugins/org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar',
        '-configuration ./config_mac',
        '-data ~/.cache/jdtls-workspace/' .. workspace_dir,
        '--add-modules=ALL-SYSTEM',
        '--add-opens java.base/java.util=ALL-UNNAMED',
        '--add-opens java.base/java.lang=ALL-UNNAMED'
  },
  root_dir = require('jdtls.setup').find_root({'.git', 'mvnw', 'gradlew'})
}
require('jdtls').start_or_attach(config)
```

**Comment by wviana** [comment, marked as answer]
I've search through the code searching for the string `quit with exit code`, but didn't find anything. Any tips?

**Reply by mfussenegger** [reply]
You're providing several options as single argument instead of as multiple arguments.

```lua
'-jar ~/Apps/jdtls/plugins/org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar',
```

Must be changed to

```lua
'-jar', '~/Apps/jdtls/plugins/org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar'
```

And the `~` in it causes it to be an invalid path, either you need expand it (e.g. by calling `vim.fn.expand` or you need to replace it with the actual path.

Same thing for `-configuration ./config_mac` and `'-data ~/.cache/jdtls-workspace/' .. workspace_dir,`

**Reply by wviana** [reply]
Hey @mfussenegger , thank you very much for helping.
I've fixed where I was passing multiple arguments and changed all paths to absolute path.
Now it's working, thank you very much.
